tracer/src/Datadog.Trace/Debugger/Expressions/MethodScopeMembers.cs

```diff
@@ -0,0 +1,51 @@
+// <copyright file="MethodScopeMembers.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+#if !NET461
+using System.Buffers;
+#endif
+
+namespace Datadog.Trace.Debugger.Expressions;
+
+internal class MethodScopeMembers
+{
+    private int _index = 0;
+
+    public MethodScopeMembers(int numberOfLocals, int numberOfArguments)
+    {
+        // 2 for 'return' and 'exception'
+#if NET461
+        Members = new ScopeMember[numberOfLocals + numberOfArguments + 2];
+#else
+        Members = ArrayPool<ScopeMember>.Shared.Rent(numberOfLocals + numberOfArguments + 2);
+#endif
+    }
+
+    public ScopeMember[] Members { get; private set; }
+
+    public Exception Exception { get; internal set; }
+
+    // we can save Return and InvocationTarget as T if we will change the native side so we will have MethodDebuggerState<T, TReturn> instead MethodDebuggerState
+    public ScopeMember Return { get; internal set; }
+
+    public ScopeMember InvocationTarget { get; internal set; }
+
+    internal void AddMember(ScopeMember member)
+    {
+        Members[_index] = member;
```

Comment:
Should we guard for `ArgumentOutOfRangeException` here, and report some more meaningful log output in case it happens?